elements/lisk-tree/src/merkle_tree.ts

```diff
@@ -11,21 +11,63 @@
  *
  * Removal or modification of this copyright notice is prohibited.
  */
+/* eslint-disable no-bitwise */
+/* eslint-disable @typescript-eslint/prefer-for-of */
 
+import { hash } from '@liskhq/lisk-cryptography';
 import { Proof } from './types';
+import { EMPTY_HASH, LEAF_PREFIX, BRANCH_PREFIX } from './constants';
 
 export class MerkleTree {
-	// eslint-disable-next-line
 	private _root: Buffer;
-	// eslint-disable-next-line
-	public constructor(_initValues: Buffer[]) {
-		this._root = Buffer.alloc(0);
+	private _dataLength = 0;
+	private _data: { [key: string]: Buffer } = {};
+
+	public constructor(initValues: Buffer[] = []) {
+		this._dataLength = initValues.length;
+		if (this._dataLength === 0) {
+			this._root = EMPTY_HASH;
+			return;
+		}
+		const leafHashes = new Array<Buffer>(this._dataLength);
+		for (let i = 0; i < this._dataLength; i += 1) {
+			const value = Buffer.concat(
+				[LEAF_PREFIX, initValues[i]],
+				LEAF_PREFIX.length + initValues[i].length,
+			);
+			const key = hash(value);
+			leafHashes[i] = key;
+			this._data[key.toString('binary')] = value;
+		}
+		const height = this._height;
```

Comment:
@shuse2 I don't see any declaration of private data member `_height`

Reply:
`height` is the getter in below =) it's calculated value.

Reply:
Then I would suggest not have a private getter which is actually calculating some value. Getters are mostly used to return private data members, which one not to expose to the public interface. 

Better to know `calcHeight`, so while reading code we know this will calculate rather returning some static value. 

Or much better option would have a particular setter for `dataLength` which also computes the height and save its static value.